tests/test_cfg_end_to_end_tests.py

```diff
@@ -175,99 +84,72 @@ def get_test_cases():
-    # to have the same structure (e.g. <lang>/<dataset>)
-    rel_path_from_root = os.path.relpath(Path(config_path).parent, DATASET_CONFIGS_ROOT)
-
-    # run data_check_fn - it will raise error if the expected test data is not found
-    data_check_fn(raw_data_dir=str(Path(test_data_root) / rel_path_from_root))
+def test_configs(config_path: str, data_check_fn: Callable, tmp_path: Path):
+    rel_path_from_root = Path(config_path).parent.relative_to(DATASET_CONFIGS_ROOT)
+    test_data_root = Path(get_e2e_test_data_path(str(rel_path_from_root)))
+    data_check_fn(raw_data_dir=str(test_data_root / rel_path_from_root))
 
-    reference_manifest = str(Path(test_data_root) / rel_path_from_root / "test_data_reference.json")
-    if not os.path.exists(reference_manifest):
+    reference_manifest = test_data_root / rel_path_from_root / "test_data_reference.json"
+    if not reference_manifest.exists():
         raise ValueError(f"Did not find reference manifest {reference_manifest}")
 
     cfg = OmegaConf.load(config_path)
     assert "processors" in cfg
-    cfg["processors_to_run"] = "all"
-    cfg["workspace_dir"] = str(tmp_path)
-    cfg["final_manifest"] = str(tmp_path / "final_manifest.json")
-    if "data_split" not in cfg:
-        cfg["data_split"] = "train"
-    cfg["processors"][0]["raw_data_dir"] = str(Path(test_data_root) / rel_path_from_root)
+    cfg.processors_to_run = "all"
+    cfg.workspace_dir = str(tmp_path)
+    cfg.final_manifest = str(tmp_path / "final_manifest.json")
+    cfg.data_split = cfg.get("data_split", "train")
+    cfg.processors[0].raw_data_dir = str(test_data_root / rel_path_from_root)
 
     run_processors(cfg)
-    # additionally, let's test that final generated manifest matches the
-    # reference file (ignoring the file paths)
-    with open(reference_manifest, "rt", encoding="utf8") as reference_fin, open(
-        cfg["final_manifest"], "rt", encoding="utf8"
-    ) as generated_fin:
-        # sorting to avoid mismatches because of randomness in utterances order
+
+    with open(reference_manifest, "rt", encoding="utf8") as reference_fin, \
+         open(cfg.final_manifest, "rt", encoding="utf8") as generated_fin:
         reference_lines = sorted(reference_fin.readlines())
         generated_lines = sorted(generated_fin.readlines())
         assert len(reference_lines) == len(generated_lines)
 
         for reference_line, generated_line in zip(reference_lines, generated_lines):
             reference_data = json.loads(reference_line)
             generated_data = json.loads(generated_line)
-            if "audio_filepath" in reference_data:
-                reference_data.pop("audio_filepath")
-                generated_data.pop("audio_filepath")
+            reference_data.pop("audio_filepath", None)
+            generated_data.pop("audio_filepath", None)
             assert reference_data == generated_data
 
-    # if CLEAN_UP_TMP_PATH is set to non-0 value, we will delete tmp_path
```

Comment:
let's add back comment

Reply:
Just trying to fix ci/cd, this will be reverted, no worries

Reply:
Done